add-ons/dialog-ui/dialogs/Dialog.js

```diff
@@ -178,6 +176,8 @@ function nullPattern() {}
 
 function removeDialog(evt) {
     if (evt.target === this.el && evt.propertyName === 'opacity') {
+        this.el.removeEventListener('transitionend', this.removeDialogBound);
```

Comment:
@Dwaynekj Added this for "completeness" -- not really necessary since the element doing the listening is about to be removed, which removes it's listeners.